docs: Add documentation for API definition compression (TT-16266) (#1330)
TykTechnologies/tyk-docs: snippets/gateway-config.mdx

Options: ["1.0", "1.1", "1.2", "1.3"].
Defaults to "1.2".

> ### storage.compress_api_definitions

Reviewer: This file is automatically generated from godocs -> I don't think this should be included in this PR, unless it's to avoid a broken link.

Looking at the code, this is different content from the comment in `config.go`

Reviewer 2: > Looking at the code, this is different content from the comment in config.go
The PR is generated by AI.

I'll remove this config section manually; this will be updated automatically when config PRs are raised

ENV: <b>TYK_GW_STORAGE_COMPRESSAPIDEFINITIONS</b><br />
Type: `bool`<br />

> When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on worker Gateways.

Reviewer: This is autogenerated from code, so I propose we leave "worker" here for now, but fix in next or a future release.

Reviewer 2: I'll update the docs now. In the future, when the config PR is raised again, I'll fix it.

> When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on worker Gateways.

Reviewer 2: ```suggestion
When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on Data Plane Gateways.
```

You can safely enable this setting on existing deployments. The Gateway continues to load previously stored uncompressed backups and stores all new backups in compressed form.

Defaults to `false`.

> 

Reviewer 2: ```suggestion
```

Options: ["1.0", "1.1", "1.2", "1.3"].
Defaults to "1.2".

### storage.compress_api_definitions
ENV: <b>TYK_GW_STORAGE_COMPRESSAPIDEFINITIONS</b><br />
Type: `bool`<br />

> When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on worker Gateways.

Reviewer 2: ```suggestion
When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on worker Gateways.

You can safely enable this setting on existing deployments. The Gateway continues to load previously stored uncompressed backups and stores all new backups in compressed form.

Defaults to `false`.
```

Defaults to "1.2".

### storage.compress_api_definitions
ENV: <b>TYK_GW_STORAGE_COMPRESSAPIDEFINITIONS</b><br />
Type: `bool`<br />

When set to `true`, enables [Zstandard (zstd)](https://facebook.github.io/zstd/) compression for API definitions stored in Redis RPC backups. This feature significantly reduces Redis memory usage in [MDCB](/api-management/mdcb) deployments where API definitions are cached locally on worker Gateways.

You can safely enable this setting on existing deployments. The Gateway continues to load previously stored uncompressed backups and stores all new backups in compressed form.

Defaults to `false`.

**Key benefits:**
- Reduces Redis memory consumption for API definition storage
- Particularly beneficial for deployments with many or large API definitions
- Compression is applied transparently when storing backups, and decompression occurs automatically when loading

**Compatibility:**
- Compressed and uncompressed data are handled automatically - the Gateway detects the format when loading backups
- Safe to enable on existing deployments; existing uncompressed backups will continue to work
- New backups will be stored compressed once enabled

> Defaults to `false`.

Reviewer 2: ```suggestion
```
